### ISSUE TITLE
Running InfoButtons using HTTPS, instead of HTTP.

### ISSUE THREAD

**Issue by DSSUSER** [logicahealth/InfoButtons#39]
We have been working with OIB for more than 2 years now, as an HTTP service, in DEV and QA environments.

We are finally ready to deploy everything to production, but we are being told by our security group, and from a SOC2 compliance aspect, that we can not use HTTP to interface with our production servers.

What are the chances of enabling HTTPS in the InfoButtons application?

NOTE: We are using the Docker container image for all of our deployments and environments.

**Comment by aniskand** [comment]
Yeah, you can run OIB in HTTPS. We use it with https in our public instance and it works just fine.

**Comment by DSSUSER** [comment]
I am trying to configure InfoButtons to use HTTPS.

We can get to the basic webpage now using HTTPS, but we are getting the following error when we attempt to login:

![InfoButtons_getUser_error](https://user-images.githubusercontent.com/15932860/103925747-fdc1c700-50e5-11eb-9440-08f9d9f56b24.JPG)

In the server.xml file, we used the following entry:

![image](https://user-images.githubusercontent.com/15932860/103926837-7aa17080-50e7-11eb-9b72-875e15348b40.png)

Is there anything else you can think of that we need to modify?

Any help in figuring out what we're doing wrong, would be greatly appreciated.

**Comment by DSSUSER** [comment]
I need to update this ticket.

BACKEND CLIENT HTTPS CALL:

The backend client is able to connect to the HTTPS service, and we are getting a response from the InfoButtons service.

However, the external calls to the document providers is going out via HTTP, instead of HTTPS.

We are not able to open the supplied URL's because the HTTPS originating request, will not allow an HTTP request, over an HTTPS connection.

Here are some of the errors that we get, when we try to access the URL / supplied link:

![image](https://user-images.githubusercontent.com/15932860/104042771-e0a1fc80-51a8-11eb-9803-e83fdbcb05af.png)

![image](https://user-images.githubusercontent.com/15932860/104043013-2eb70000-51a9-11eb-8387-f19b14a785cd.png)

![image](https://user-images.githubusercontent.com/15932860/104043582-024fb380-51aa-11eb-9b4b-0b41a735d662.png)

![image](https://user-images.githubusercontent.com/15932860/104043661-227f7280-51aa-11eb-9fa2-16347a71bcbd.png)

--------

SITE LITE GUI LOGIN:

We can get to the GUI Login screen, but as you can see from the errors below, the "getUser" function is being called, using an HTTP request.

Our HTTPS call will not allow mis-matched HTTPS and HTTP requests to be mixed in the same request.

![image](https://user-images.githubusercontent.com/15932860/104042043-ef3be400-51a7-11eb-854e-d08fcab2cc3b.png)

![image](https://user-images.githubusercontent.com/15932860/104041937-c9164400-51a7-11eb-91bd-d216bc74a1ac.png)

------

How can we get rid of this mismatch?

Note, that if I take any of the supplied URLs, and simply replace HTTP with HTTPS, they all work!

I can access all of the HTTP pages supplied, by simply converting them to HTTPS.

Again, if we tell the application to use HTTPS, how can we ensure that all subsequent calls use HTTPS instead of HTTP?

**Comment by aniskand** [comment]
I'm not sure how we handle this in our environments, I think we have httpd configured to redirect all requests to https. I'm going to update the Docker image to fix these issues as well as include my new code for handling the changes to UTS authentication .

**Comment by DSSUSER** [comment]
Hi Andrew,

Thank you very much.

Going forward, for security purposes, we will probably use HTTPS in all of our environments.

It does not make sense for us to use HTTP in DEV and QA, if we need HTTPS in production.

As such, if you want to convert the code to always use HTTPS, we would definitely not object.

Have a great weekend.

**Comment by DSSUSER** [comment]
Good morning Andrew,

I was just checking in, to see if you have any kind of status update yet.

Have a great day.

**Comment by aniskand** [comment]
I'm hoping I can get to this before the end of the week.

**Comment by DSSUSER** [comment]
Thanks Andrew.

Just a reminder that the UMLS UTS site will officially switch to the new login method this Friday, the 15th.

**Comment by DSSUSER** [comment]
GM Andrew,

Do you have any updates for this issue?

**Comment by DSSUSER** [comment]
For testing purposes, does your public deployment of InfoButtons use HTTPS?

If so, until we can get our internal HTTPS version working, can we point to your public version?

If yes, what is that URL?

**Comment by gdelfiol** [comment]
We do support https on our public server and you are welcome to use it. Here is the base url:

https://service.oib.utah.edu/infobutton-service/infoRequest?

**Comment by DSSUSER** [comment]
We pointed our application to your public interface.

We are getting the following error:

![image](https://user-images.githubusercontent.com/15932860/105406803-ec56df80-5bfa-11eb-894b-624ad2673649.png)

**Comment by gdelfiol** [comment]
Can you send a couple sample URLs generated by your application?

**Comment by DSSUSER** [comment]
Please ignore the previous post. 

Our developer made a change a few days ago that caused the error.

Everything is working as expected right now.

**Comment by gdelfiol** [comment]
Perfect, let us know if you need any help.

**Comment by DSSUSER** [comment]
Andrew was supporting a Docker container image with InfoButtons completely configured.

Is anyone there able to support this Docker image?

Is anyone there able to modify the Docker image to use HTTPS instead of HTTP?

Is anyone there able to modify the Docker image to incorporate the new UMLS UTS login requirements that went into effect on the 15th of January?

Is anyone there able to modify the Docker image to incorporate the same changes that went into the main InfoButtons configuration?

**Comment by DSSUSER** [comment]
Are there any updates for this issue?

Are there any updates for this issue, as they pertain to the Docker image?
